src/test/java/co/cask/cdc/plugins/sink/CDCBigTableConfigTest.java

```diff
@@ -0,0 +1,134 @@
+package co.cask.cdc.plugins.sink;
+
+import co.cask.cdap.etl.api.validation.InvalidConfigPropertyException;
+import com.google.bigtable.repackaged.com.google.cloud.ServiceOptions;
+import org.assertj.core.api.Assertions;
+import org.junit.Assume;
+import org.junit.Test;
+
+public class CDCBigTableConfigTest {
+  private static final String VALID_REF = "test-ref";
+  private static final String VALID_PROJECT = "test-project";
+  private static final String VALID_INSTANCE = "test-instance";
+  private static final String VALID_ACCOUNT_FILE_PATH
+    = CDCBigTableConfigTest.class.getResource("/credentials.json").getPath();
+
+  @Test
+  public void testValidateValidConfig() {
+    CDCBigTableConfig config = new CDCBigTableConfig(
+      VALID_REF,
+      VALID_INSTANCE,
+      VALID_PROJECT,
+      VALID_ACCOUNT_FILE_PATH
+    );
+
+    Assertions.assertThatCode(config::validate)
+      .doesNotThrowAnyException();
+  }
+
+  @Test
+  public void testValidateReference() {
+    CDCBigTableConfig config = new CDCBigTableConfig(
+      "",
+      VALID_INSTANCE,
+      VALID_PROJECT,
+      VALID_ACCOUNT_FILE_PATH
+    );
+
+    Assertions.assertThatThrownBy(config::validate)
+      .isInstanceOf(InvalidConfigPropertyException.class);
+  }
+
+  @Test
+  public void testValidateMissingCredentialsFile() {
+    CDCBigTableConfig config = new CDCBigTableConfig(
+      VALID_REF,
+      VALID_INSTANCE,
+      VALID_PROJECT,
+      "/tmp/non_existing_file"
+    );
+
+    Assertions.assertThatThrownBy(config::validate)
+      .isInstanceOf(InvalidConfigPropertyException.class);
+  }
+
+  @Test
+  public void testValidateMissingProjectId() {
+    Assume.assumeTrue(ServiceOptions.getDefaultProjectId() == null);
+
+    CDCBigTableConfig config = new CDCBigTableConfig(
+      VALID_REF,
+      VALID_INSTANCE,
+      null,
+      VALID_ACCOUNT_FILE_PATH
+    );
+
+    Assertions.assertThatThrownBy(config::validate)
+      .isInstanceOf(InvalidConfigPropertyException.class);
+  }
+
+  @Test
+  public void testValidateMissingInstanceId() {
+    CDCBigTableConfig config = new CDCBigTableConfig(
+      VALID_REF,
+      null,
+      VALID_PROJECT,
+      VALID_ACCOUNT_FILE_PATH
+    );
+
+    Assertions.assertThatThrownBy(config::validate)
+      .isInstanceOf(InvalidConfigPropertyException.class);
+  }
+
+  @Test
+  public void testResolveProjectId() {
+    CDCBigTableConfig config = new CDCBigTableConfig(
+      VALID_REF,
+      VALID_INSTANCE,
+      null,
+      VALID_ACCOUNT_FILE_PATH
+    );
+
+    Assertions.assertThat(config.resolveProject())
+      .isEqualTo(ServiceOptions.getDefaultProjectId());
```

Comment:
Tests should use plain junit assertions when possible. Does this give you anything more than normal junit Assert.assertEquals()?

Reply:
> just noticed that a lot of files are missing copyright. This is supposed to be in the rat check, have we not set it up for this project?

This plugin was not configured for this project as well as for google-cloud. I may submit another PR with rat plugin and updated copyrights in all classes.

Reply:
Actually assertEquals is almost the only case where JUnit assertion works in same way as AssertJ one. And even in this case there are some differences:
- JUnit assertions provokes developer to pass "expected" and "actual" params in wrong order and you'll have confusing failure message. This is fine when test failure is reproducible but often leads to loss of time when bug is reproducible only on CI or once per week (for example).
- JUnit's assertEquals does not have custom representation, custom comparator and soft assertions support.
- JUnit's assertEquals is less smart then one from AssertJ in some cases. For example: if you compare numbers with floating point assertJ will use comparison delta by default.

I also wanted to add why I prefer AssertJ in general:
- it provides much better failure messages. For example with JUnit:
    - you often use assertNotNull that will fail with AssertionError with no message (no details at all)
    - with code like Assert.assertTrue(one.getSchema().getFields().size() == two.getSchema().getFields().size()) you will not see wrong actual schema to analyze what happened. This is a problem when bug is not reproducible.
- it is single library that covers all possible cases where you previously needed Junit, TestNG or Hamcrest assertion libraries.
- it makes code much more readable and concise.
- it provides custom assertions functionality and other extendability options that allows to make testing frameworks. That approach reduces code duplication and makes all tests look similar without complex guidelines.
- it gives great auto-completion for IDE which is very handy with custom assertions.
- more detailed key difference overview: https://professional-practical-programmer.com/2016/06/26/assert-j/
- AsserJ features highlight: http://joel-costigliola.github.io/assertj/assertj-core-features-highlight.html

In summary. There are definitely some cases where AssertJ does better job than Junit assertions. And in this case there is a choice: to use 3 libraries depending on use case (Junit, hamcrest, AssertJ) or to use only 1 library in all cases (AssertJ). As for me, second approach is easier to describe in guides and to understand than trying to describe what library should be used in what case.

Please, consider this as my personal opinion and just a proposal. Thank you for your time.

Reply:
I agree that junit doesn't have the most features, but I think other more powerful testing frameworks can sometimes allow you to get away with not having the best abstractions in the actual code. So I hesitate to add it where it is not necessary. As a general rule, I think it makes sense to start with only JUnit, since that is the standard across the other repositories, and only add others if there is a compelling reason to do it.

Reply:
fixed